### ISSUE TITLE
Default skipDefs value skips checking annotated anonymous class

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#245]
```
What steps will reproduce the problem?
1. Create this class:

import org.eclipse.jdt.annotation.NonNull;
import org.eclipse.jdt.annotation.Nullable;

public class Snippet {
  public static void main(String[] args) {
    call(new /*@Nullable*/Runnable() {
      @Override public void run() {
        @Nullable Object veryNull = null;
        @NonNull Object notNull = veryNull;
        notNull.toString();
      }
    });
  }

  private static void call(Runnable r) {
    r.run();
  }
}

2. Run nullness checker on it

What is the expected output? What do you see instead?
The class should not typecheck with (assignment.type.incompatible) error 
message. Instead no errors are emitted.

What version of the product are you using? On what operating system?
Checkers Framework 1.6.6

Please provide any additional information below.
The missed error is caused by a combination of several factors:

1. The default regex pattern for skipDefs is "\(" which normally doesn't match 
any class.
2. Neither SourceChecker.shouldSkipDefs(ClassTree) nor 
InternalUtils.typeOf(Tree) retrieve the underlying type from an annotated type.
3. SourceChecker.shouldSkipDefs(ClassTree) constructs the type name to match 
against by calling toString() on the type.
4. AnnotatedType uses a debug output format which happens to include characters 
like ():{}

```

Original issue reported on code.google.com by `timo.kin...@gmail.com` on 21 Aug 2013 at 3:29
